lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/DriverStatusHTTPHandler.java

```diff
@@ -126,4 +157,25 @@ String waitAndGetMessage() {
   static String getMessageForStatus(final ReefServiceProtos.JobStatusProto status) {
     return status.getState().name();
   }
+
+  /**
+   * Schedules an alarm, if needed.
+   * <p>
+   * The alarm will prevent the Clock from going idle. This gives the .NET Client time to make a call to this HTTP
+   * handler.
+   */
+  private void scheduleAlarm() {
+    if (this.wasCalledViaHTTP || retry >= MAX_RETRIES) {
+      return; // No alarm necessary anymore.
```

Comment:
Shall we write something to the log before `return`?

Reply:
Good idea. Done.